lib/participant.js

```diff
@@ -424,4 +535,26 @@ function reemitTrackEvents(participant, track) {
   participant._trackEventReemitters.set(track.id, trackEventReemitters);
 }
 
+/**
+ * Re-emit {@link TrackPublication} events.
+ * @param {Participant} participant
+ * @param {TrackPublication} publication
+ */
+function reemitTrackPublicationEvents(participant, publication) {
+  const publicationEventReemitters = new Map();
+
+  if (participant.state === 'disconnected') {
+    return;
+  }
+
+  participant._getTrackPublicationEvents().forEach(([publicationEvent, participantEvent]) => {
+    publicationEventReemitters.set(publicationEvent, (...args) => {
+      participant.emit(participantEvent, ...args, publication);
```

Comment:
Will this clash withe Track-level events that are already emitted?

Reply:
@markandrus I don't think we are re-emitting any `TrackPublication` events that are also fired on `Track`s. Right now, we only re-emit `subscriptionFailed`.